test/TabContainerSpec.js

```diff
@@ -308,6 +314,46 @@ describe('<TabContainer>', () => {
     instance.single('[eventKey=0]').single('.active');
   });
 
+  [
+    [[1, 2], [2, 1]],
+    [[2, 1], [1, 2]],
+  ].forEach(([order1, order2]) => {
+    it('should handle event key swaps', () => {
```

Comment:
@jquense 

Never mind, I found another tab animation edge case (the logic in the previous commit would drop the animation if this happened... the logic on master would instead fail the duplicate `eventKey` invariant)

basically doing this type of imperative animation end callback handling in React is the worst thing ever. I don't think even not using context would simplify this appreciably.